snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedEditingContext.java

```diff
@@ -542,300 +549,114 @@ private Concept getConceptChecked(final CDOID cdoId) {
+			tryDelete((Description) object, force);
 		} else if (object instanceof Relationship) {
-			delete((Relationship) object, force);
+			tryDelete((Relationship) object, force);
 		} else if (object instanceof SnomedRefSet) {
-			delete((SnomedRefSet) object, force);
+			tryDelete((SnomedRefSet) object, force);
 		} else if (object instanceof SnomedRefSetMember) {
-			delete((SnomedRefSetMember) object, force);
+			tryDelete((SnomedRefSetMember) object, force);
 		} else {
 			super.delete(object, force);
 		}
-	}
-	
-	private void delete(Concept concept, boolean force) {
-		deletionPlan = canDelete(concept, deletionPlan, force);
-		if(deletionPlan.isRejected()) {
-			throw new ConflictException(deletionPlan.getRejectionReasons().toString());
-		}
-	}
-
-	private void delete(Description description, boolean force) {
-		deletionPlan = canDelete(description, deletionPlan, force);
-		if(deletionPlan.isRejected()) {
-			throw new ConflictException(deletionPlan.getRejectionReasons().toString());
-		}
-	}
-
-	private void delete(Relationship relationship, boolean force) {
-		deletionPlan = canDelete(relationship, deletionPlan, force);
-		if(deletionPlan.isRejected()) {
-			throw new ConflictException(deletionPlan.getRejectionReasons().toString());
-		}
-	}
-
-	private void delete(SnomedRefSet refSet, boolean force) {
-		deletionPlan = canDelete(refSet, deletionPlan, force);
-		if (deletionPlan.isRejected()) {
-			throw new ConflictException(deletionPlan.getRejectionReasons().toString());
-		}
-	}
-	
-	private void delete(SnomedRefSetMember member, boolean force) {
-		deletionPlan = canDelete(member, deletionPlan, force);
+		
 		if (deletionPlan.isRejected()) {
-			throw new ConflictException(deletionPlan.getRejectionReasons().toString());
+			throw new ConflictException(deletionPlan.toString());
 		}
 	}
 	
-	public SnomedDeletionPlan canDelete(Concept concept, SnomedDeletionPlan deletionPlan, boolean force) {
-		if (deletionPlan == null) {
-			deletionPlan = new SnomedDeletionPlan();
-		}
+	private void tryDelete(Concept concept, boolean force) {
 		
 		// Check if concept is already released, and this is not a forced delete
 		if (concept.isReleased() && !force) {
 			deletionPlan.addRejectionReason(String.format(COMPONENT_IS_RELEASED_MESSAGE, "concept", toString(concept)));
-			return deletionPlan;
+			return;
 		}
 		
-		// Also check inbound relationships, as these could have been released with different effective times
 		for (Relationship relationship : getInboundRelationships(concept.getId())) {
+			
 			if (relationship != null) {
-				deletionPlan = canDelete(relationship, deletionPlan, force);
-				if (deletionPlan.getRejectionReasons().size() > 0) {
-					deletionPlan.addRejectionReason(String.format(UNABLE_TO_DELETE_CONCEPT_MESSAGE, toString(concept)));
-					return deletionPlan;
+				
+				tryDelete(relationship, force);
+				
+				if (deletionPlan.isRejected()) {
+					deletionPlan.addRejectionReason(
```

Comment:
Each released inbound relationship now results in two rejection messages, the regular "relationship is released" as well as "concept has released inbound relationship" one. Good, bad?

Reply:
I think this is fine because it will make it more clear why it failed when we log the rejection reasons.